### PR TITLE
fix(security): bounds-check integer conversion in llm client

### PR DESCRIPTION
**Pull request by IsmaelMartinez**
## Summary

Addresses CodeQL alert `go/incorrect-integer-conversion` (high) at `internal/llm/client.go:63`.

`SetDailyLimit` converted an `int` argument directly to `int32` via `int32(limit)`. On 64-bit platforms `int` is 64-bit, so values above `math.MaxInt32` would silently wrap and negative values could roll over too. Added a bounds-check that clamps oversized values to `math.MaxInt32` and negative values to `0` (the existing "unlimited" sentinel) before storing into the `atomic.Int32`.

## Test plan

- [x] `go build ./...`
- [x] `go test ./...` (all packages pass)
- [ ] CodeQL re-scan confirms alert closes

Co-Authored-By: Claude Opus 4.7 (1M context) <noreply@anthropic.com>

### CODE REVIEW

**Review by IsmaelMartinez — COMMENTED** [review]
Reviewed the clamp logic in `SetDailyLimit`:

- `limit < 0` → stored as 0 (matches existing 'unlimited' sentinel; `checkAndIncrementDaily` treats `limit <= 0` as unlimited, so this is consistent).
- `limit > math.MaxInt32` → stored as `math.MaxInt32`.
- In-range values pass through unchanged (no behaviour change for normal config).
- No off-by-one (boundary value `MaxInt32` itself flows through the default branch).

`go vet ./...` clean, `go test ./internal/llm/...` passes. CodeQL alert `go/incorrect-integer-conversion` should close on next scan. Looks good to me.

File: internal/llm/client.go [hunk 1]
```diff
@@ -59,8 +60,19 @@ func New(apiKey string, logger *slog.Logger) *Client {
...
+	switch {
+	case limit < 0:
+		stored = 0
+	case limit > math.MaxInt32:
+		stored = math.MaxInt32
+	default:
+		stored = int32(limit)
+	}
+	c.dailyLimit.Store(stored)
 }
```
[the remark this answers is not in the document]
Comment (IsmaelMartinez) [reply internal/llm/client.go]: Good catch — confirmed the overflow path: with limit=MaxInt32, once `dailyCount` reaches MaxInt32 the next `Add(1)` wraps to MinInt32, `count > limit` reads false, the rollback is skipped, and the cap is silently bypassed.

Applied your suggested fix in 5d39b12: clamp upper bound to `math.MaxInt32-1` and switched to the built-in `min`/`max` helpers. Kept `int32` for now to avoid expanding the change beyond the CodeQL alert; happy to follow up with an int64 refactor in a separate PR if you think it's worth it.

**Comment by IsmaelMartinez** [comment]
Addressed Gemini Code Assist's review comment in 5d39b12: clamped upper bound to `math.MaxInt32-1` (closing the counter-overflow bypass) and adopted the built-in `min`/`max` helpers for clarity. `go vet ./...` clean and llm tests pass locally. Did not pursue the int64 refactor in this PR to keep the diff scoped to the CodeQL alert; can do it in a follow-up.
